packages/plugin-commands-script-runners/src/run.ts

```diff
@@ -134,6 +134,11 @@ export async function handler (
 ) {
   let dir: string
   const [scriptName, ...passedThruArgs] = params
+  // For backward compatibility
+  const firstDoubleDash = passedThruArgs.findIndex((arg) => arg === '--')
```

Comment:
I pattern matched this from the same change in `pnpm exec`: https://github.com/pnpm/pnpm/pull/3492

https://github.com/pnpm/pnpm/blob/0355d30e62e4de0c045b6bb3f4fa7dcd6836f33b/packages/plugin-commands-script-runners/src/exec.ts#L76-L79

pnpm v6 allowed `--` anywhere in the `args` portion of `pnpm run <command> [..args]`, so this extra `.findIndex` logic tries to maintain full backwards compatible behavior for that. Although I feel like we should just swallow the first `--` like `pnpm exec` does.

I could see the behavior in this scenario being a bit surprising:

```
❯ pnpm run --silent echo a -- b
a b
```

You might expect:

```
❯ pnpm run --silent echo a -- b
a -- b
```

Reply:
Although the behavior above didn't change in this PR, I do worry this will come back to bite someone that thinks `--` can be dropped entirely after this change. Just a heads up @zkochan.

Happy to make any additional followup changes before the pnpm v7 release if we want to.

Reply:
I think it is fine if we don't swallow `--` at all. This is a breaking change in v7. So it won't be a surprise to anyone.

Reply:
👍 Cool. I'll open a pull request reverting this block later tonight when I'm near my computer.